File: CompilationTest/ExplicitGenericTypeTest/results/ExplicitGenericTypeTest.pl
use strict;
use warnings;

package TestClass;

sub new
{
    my $class = shift;
    my $self = {};
    bless $self, $class;
    return $self;
}

sub test_method {
    my ( $self ) = @_;
    my $result = ["y"];
    my $map = {
      x => 5,
    };
    my $keys = keys %{$map};
    print((${$result}[0]) . "\n");
    print((${$keys}[0]) . "\n");
}

package Program;

eval {
    my $c = new TestClass();
    $c->test_method();
};
if ($@) {
    print "Exception: " . $@
}

File: CompilationTest/ExplicitGenericTypeTest/results/ExplicitGenericTypeTest.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

class TestClass {
    public void testMethod() throws Exception
    {
        List<String> result = new ArrayList<String>(Arrays.asList("y"));
        HashMap<String, Integer> map = new HashMap<String, Integer>();
        map.put("x", 5);
        List<String> keys = new ArrayList(map.keySet());
        System.out.println(result.get(0));
        System.out.println(keys.get(0));
    }
}

class Program {
    public static void main(String[] args) throws Exception {
        try {
            new TestClass().testMethod();
        } catch (Exception err) {
            System.out.println("Exception: " + err.getMessage());
        }
    }
}